fastmcp-auth/fastmcp_auth.py

```
@@ -0,0 +1,330 @@
+                token=token,
+                client_id=client_id,
+                email=email,
+                scopes=scopes,
+                expires_at=expires_at,
+                issuer=payload.get('iss'),
+                audience=payload.get('aud'),
+                additional_claims={k: v for k, v in payload.items() 
+                                 if k not in ['sub', 'client_id', 'email', 'scope', 'scopes', 'exp', 'iss', 'aud', 'iat']}
+            )
+        except jwt.InvalidTokenError as e:
+            raise jwt.InvalidTokenError(f"Invalid token: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)